ReactiveCocoa/NSObject+Intercepting.swift

```diff
@@ -6,30 +6,436 @@ extension Reactive where Base: NSObject {
 	/// Create a signal which sends a `next` event at the end of every invocation
 	/// of `selector` on the object.
 	///
+	/// It completes when the object deinitializes.
+	///
+	/// - note: Observers to the resulting signal should not call the method
+	///         specified by the selector.
+	///
 	/// - parameters:
 	///   - selector: The selector to observe.
 	///
 	/// - returns:
 	///   A trigger signal.
 	public func trigger(for selector: Selector) -> Signal<(), NoError> {
-		return base.synchronized {
-			let map = associatedValue { _ in NSMutableDictionary() }
+		return setupInterception(base, for: selector).map { _ in }
+	}
+
+	/// Create a signal which sends a `next` event, containing an array of bridged
+	/// arguments, at the end of every invocation of `selector` on the object.
+	///
+	/// It completes when the object deinitializes.
+	///
+	/// - note: Observers to the resulting signal should not call the method
+	///         specified by the selector.
+	///
+	/// - parameters:
+	///   - selector: The selector to observe.
+	///
+	/// - returns:
+	///   A signal that sends an array of bridged arguments.
+	public func signal(for selector: Selector) -> Signal<[Any?], NoError> {
+		return setupInterception(base, for: selector).map(unpackInvocation)
+	}
+}
+
+
+// A summary of associated values involved.
+//
+// Runtime subclasses:
+// - `&AssociationKey.intercepted` (ObjC+RuntimeSubclassing.swift)
+//   Nullable. Exists only in runtime subclasses generated by external parties.
+//   `true` indicates the runtime subclass has already been swizzled.
+//
+// - `&AssociationKey.intercepted` (This file)
+//   Nullable. `true` indicates the runtime subclass has already been prepared for method
+//   interception.
+//
+// - `&AssociationKey.signatureCache`
```

Comment:
These should go on the properties themselves. (Then they'll also be viewable in Xcode directly at the usage site.)